packages/apm/src/integrations/tracing.ts

```diff
@@ -14,6 +14,73 @@ import { Span as SpanClass } from '../span';
 import { SpanStatus } from '../spanstatus';
 import { Transaction } from '../transaction';
 
+/** Holds the latest LargestContentfulPaint value (it changes during page load). */
+let lcp: { [key: string]: any };
+
+/** Force any pending LargestContentfulPaint records to be dispatched. */
+let forceLCP = () => {
+  /* No-op, replaced later if LCP API is available. */
+};
+
+// Based on reference implementation from https://web.dev/lcp/#measure-lcp-in-javascript.
+{
+  // Use a try/catch instead of feature detecting `largest-contentful-paint`
+  // support, since some browsers throw when using the new `type` option.
+  // https://bugs.webkit.org/show_bug.cgi?id=209216
+  try {
+    // Keep track of whether (and when) the page was first hidden, see:
+    // https://github.com/w3c/page-visibility/issues/29
+    // NOTE: ideally this check would be performed in the document <head>
+    // to avoid cases where the visibility state changes before this code runs.
+    let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+    document.addEventListener(
+      'visibilitychange',
+      event => {
+        firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+      },
+      { once: true },
+    );
+
+    const updateLCP = (entry: PerformanceEntry) => {
+      // Only include an LCP entry if the page wasn't hidden prior to
+      // the entry being dispatched. This typically happens when a page is
+      // loaded in a background tab.
+      if (entry.startTime < firstHiddenTime) {
+        // NOTE: the `startTime` value is a getter that returns the entry's
+        // `renderTime` value, if available, or its `loadTime` value otherwise.
+        // The `renderTime` value may not be available if the element is an image
+        // that's loaded cross-origin without the `Timing-Allow-Origin` header.
+        lcp = {
+          // @ts-ignore
+          elementId: entry.id,
+          // @ts-ignore
+          elementSize: entry.size,
```

Comment:
These are not always available. We may or may not want more details from the LCP entry. I've put those here as examples of what is possible.